### PR TITLE
chore: move sdkLanguage over to the protocol init call

### LINKED ISSUE #7741: [Bug]: Playwright executable path not found message should be cross-language compatible
Problem: We currently show on Python etc. npx playwright install when the executables are not installed.

https://github.com/microsoft/playwright/blob/b56e2b713a6ec4b95faaa9d6e3ced9bb538c4c2e/src/utils/registry.ts#L252-L259

Needs protocol changes. Since we should notify the server early on which language we are. Either when we launch or better before. Currently its done on newContext.

Environment variables are not the best there, since if we do it over WS, then its not working.

### PR DESCRIPTION
**Pull request by mxschmitt**
Fixes #7741

### CODE REVIEW

File: src/server/chromium/crPage.ts [hunk 1]
```diff
@@ -848,18 +848,12 @@ class FrameSession {
     const ffmpegPath = registry.findExecutable('ffmpeg')!.executablePath();
     // TODO: use default error message once it's ready.
     if (!ffmpegPath || !canAccessFile(ffmpegPath)) {
```
Comment (dgozman) [inline src/server/chromium/crPage.ts:850]: Why don't we use executablePathOrDie() here? We can append `Please install ffmpeg in order to record video.` to the error, or pass custom text to `executablePathOrDie` instead. It already has a nice `<3 Playwright Team` message.

File: src/server/playwright.ts [hunk 2]
```diff
@@ -54,6 +55,6 @@ export class Playwright extends SdkObject {
   }
 }
 
-export function createPlaywright(isInternal = false) {
-  return new Playwright(isInternal);
+export function createPlaywright(isInternal: boolean = false, sdkLanguage: string = 'javascript') {
```
Comment (dgozman) [inline src/server/playwright.ts:58]: I'd prefer sdkLanguage to be required.

File: src/utils/registry.ts [hunk 3]
```diff
@@ -660,6 +660,18 @@ function markerFilePath(browserDirectory: string): string {
   return path.join(browserDirectory, 'INSTALLATION_COMPLETE');
 }
 
+export function buildPlaywrightCLICommand(cliLanguage: string, parameters: string): string {
+  const cliLanguageToCLIBuilder: Partial<Record<string, (parameters: string) => string>> = {
```
Comment (dgozman) [inline src/utils/registry.ts:664]: Why Partial?

File: src/utils/registry.ts [hunk 4]
```diff
@@ -660,6 +660,18 @@ function markerFilePath(browserDirectory: string): string {
...
+    'python': (parameters: string) => `playwright ${parameters}`,
+    'java': (parameters: string) => `mvn exec:java -e -Dexec.mainClass=com.microsoft.playwright.CLI -Dexec.args="${parameters}"`,
+    'csharp': (parameters: string) => `playwright ${parameters}`,
+    'javascript': (parameters: string) => `npx playwright ${parameters}`,
+  };
+  const build = cliLanguageToCLIBuilder[cliLanguage];
+  assert(build, `Unknown CLI language: ${cliLanguage}`);
```
Comment (dgozman) [inline src/utils/registry.ts:671]: Let's fallback to javascript instead.

File: src/dispatchers/dispatcher.ts [hunk 5]
```diff
@@ -125,16 +125,16 @@ export type DispatcherScope = Dispatcher<any, any>;
 export class Root extends Dispatcher<{ guid: '' }, {}> {
   private _initialized = false;
 
-  constructor(connection: DispatcherConnection, private readonly createPlaywright?: (scope: DispatcherScope) => Promise<PlaywrightDispatcher>) {
+  constructor(connection: DispatcherConnection, private readonly createPlaywright?: (scope: DispatcherScope, options: channels.RootInitializeParams) => Promise<PlaywrightDispatcher>) {
```
Comment (dgozman) [inline src/dispatchers/dispatcher.ts:128]: Let's rename language to sdkLanguage for consistency.

File: src/server/chromium/crPage.ts (hunk 1 above)
Comment (mxschmitt) [reply src/server/chromium/crPage.ts:850]: good catch! one less exported function.

File: src/utils/registry.ts (hunk 3 above)
Comment (mxschmitt) [reply src/utils/registry.ts:664]: This ends up in a `string` type otherwise. I rewrote this part, so not needed anymore.

**Review by dgozman — APPROVED** [review]

**Pull request merged**